src/BenchmarkDotNet.Diagnostics.Windows/Sessions.cs

```diff
@@ -43,21 +58,21 @@ public KernelSession(DiagnoserActionParameters details, EtwProfilerConfig config
             : base(KernelTraceEventParser.KernelSessionName, details, config, creationTime)
         {
         }
-        
+
         protected override string FileExtension => ".kernel.etl";
 
         internal override Session EnableProviders()
         {
-            var keywords = Config.KernelKeywords 
-                | KernelTraceEventParser.Keywords.ImageLoad // handles stack frames from native modules, SUPER IMPORTANT! 
+            var keywords = Config.KernelKeywords
+                | KernelTraceEventParser.Keywords.ImageLoad // handles stack frames from native modules, SUPER IMPORTANT!
                 | KernelTraceEventParser.Keywords.Profile; // CPU stacks
 
             if (Details.Config.GetHardwareCounters().Any())
                 keywords |= KernelTraceEventParser.Keywords.PMCProfile; // Precise Machine Counters
 
             try
             {
-                TraceEventSession.EnableKernelProvider(keywords, KernelTraceEventParser.Keywords.Profile);
+                TraceEventSession.EnableKernelProvider(keywords, keywords); //TODO ? In this way it is created in PerfView
```

Comment:
regarding the "TODO" comment: do we need this change? does it make any difference?

Reply:
Thanks for this comment!

 I checked it and the second parameter means:
`stackCapture = Specifies which events should have their stack traces captured when an event is logged`
After my change, all kernelKeywords from the configuration would have a captured stack. So I reverted this change. 

Thanks to this comment I checked what more I could cut. In the last commit, I sent some optimization.
Currently, for the user session, BDN collects only information about `EngineEventSource`. 
For kernel session I collect  `VirtualAlloc` and `VAMap`:
```diff
-            var kernelKeywords = KernelTraceEventParser.Keywords.Default | KernelTraceEventParser.Keywords.VirtualAlloc | KernelTraceEventParser.Keywords.VAMap;
+            var kernelKeywords = KernelTraceEventParser.Keywords.VirtualAlloc | KernelTraceEventParser.Keywords.VAMap;
```
`KernelSession` adds some extra kernel keywords 
https://github.com/dotnet/BenchmarkDotNet/blob/ec296dc45de798f7407852d5ab7febe2b457eca4/src/BenchmarkDotNet.Diagnostics.Windows/Sessions.cs#L51-L53
and in this case, it could be removed. But I think that it could be improved in the next PR becouse it touch `EtwProfiler` and `ConcurrencyVisualizerProfiler`.